File: src/main/kotlin/com/example/onlineShopOnKotlin/security/UserRepository.java
package com.example.onlineShopOnKotlin.security;

import com.example.onlineShopOnKotlin.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public class UserRepository extends JpaRepository<User, Long> {
}

File: src/main/kotlin/com/example/onlineShopOnKotlin/repository/UserRepository.kt
package com.example.onlineShopOnKotlin.repository

import com.example.onlineShopOnKotlin.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface UserRepository: JpaRepository<User, Long> {
    fun findByEmail(email:String):User
    fun findByName(name:String): Optional<User>
}